### PR TITLE
Kitchenflow workaround

### PR DESCRIPTION
**Pull request by AntonGerasimovich**

### CODE REVIEW

File: app/src/main/java/com/example/kitchenflow/ui/main/MainFragment.kt [hunk 1]
```diff
@@ -68,6 +82,28 @@ class MainFragment : Fragment() {
             }
             adapter = OrdersAdapter(requireContext())
             ordersListRv.adapter = adapter
+            dateTv.text =
+                SimpleDateFormat("EEE, MMMM dd", Locale.ENGLISH).format(Calendar.getInstance().time)
+            currentTv.text = "${currentTv.text} (${viewModel.orders.value?.size})"
```
Comment (NikitaLychkovski) [inline app/src/main/java/com/example/kitchenflow/ui/main/MainFragment.kt]: should be inside viewmodel

File: app/src/main/java/com/example/kitchenflow/ui/main/MainFragment.kt [hunk 2]
```diff
@@ -68,6 +82,28 @@ class MainFragment : Fragment() {
...
+                override fun onTextChanged(s: CharSequence?, start: Int, before: Int, count: Int) {
+                    val filter = s.toString()
+                    if (filter.isNotEmpty()) {
+                        viewModel.orders.value?.filter { filter in it.username || filter in it.shortId }
+                            ?.let { adapter.addAll(it) }
+                    } else viewModel.orders.value?.let { adapter.addAll(it) }
+                }
+
+                override fun afterTextChanged(s: Editable?) {}
+            })
```
Comment (NikitaLychkovski) [inline app/src/main/java/com/example/kitchenflow/ui/main/MainFragment.kt]: same should be inside viewmodel

File: app/src/main/java/com/example/kitchenflow/ui/main/OrdersAdapter.kt [hunk 3]
```diff
@@ -53,6 +55,43 @@ class OrdersAdapter(
...
+        var orderPrepTime =
+            order.scheduledFor.convertToMilliseconds() - order.preparationTimeSec * 1000 + KITCHEN_PREP_MILLIS
+        if (order.orderType == OrderType.Curbside || order.orderType == OrderType.Takeout) {
+            ioTimerTv.text = if (currentTime >= orderPrepTime) {
+                ioTimerTv.setTextColor(context.getColor(R.color.colorPrimaryRed))
+                order.timer = ((currentTime - orderPrepTime) / 1000 / 60).toInt()
+                "+${if (order.timer >= 99) 99 else order.timer}m"
+            } else {
+                "${((orderPrepTime - currentTime) / 1000 / 60).toInt()}m"
+            }
```
Comment (NikitaLychkovski) [inline app/src/main/java/com/example/kitchenflow/ui/main/OrdersAdapter.kt]: This code is the same block as code below. Move it to the function

File: app/src/main/java/com/example/kitchenflow/ui/main/OrdersAdapter.kt [hunk 4]
```diff
@@ -53,6 +55,43 @@ class OrdersAdapter(
         }
     }
 
+    private fun ItemOrderBinding.setUpTimer(order: OrderModel) {
+        val currentTime = System.currentTimeMillis()
+        var orderPrepTime =
+            order.scheduledFor.convertToMilliseconds() - order.preparationTimeSec * 1000 + KITCHEN_PREP_MILLIS
```
Comment (NikitaLychkovski) [inline app/src/main/java/com/example/kitchenflow/ui/main/OrdersAdapter.kt]: `order.scheduledFor ` Why you are storing a date as a string?

File: app/src/main/java/com/example/kitchenflow/ui/main/OrdersAdapter.kt [hunk 5]
```diff
@@ -53,6 +55,43 @@ class OrdersAdapter(
...
+        if (order.orderType == OrderType.Curbside || order.orderType == OrderType.Takeout) {
+            ioTimerTv.text = if (currentTime >= orderPrepTime) {
+                ioTimerTv.setTextColor(context.getColor(R.color.colorPrimaryRed))
+                order.timer = ((currentTime - orderPrepTime) / 1000 / 60).toInt()
+                "+${if (order.timer >= 99) 99 else order.timer}m"
```
Comment (NikitaLychkovski) [inline app/src/main/java/com/example/kitchenflow/ui/main/OrdersAdapter.kt]: `m"` this part of the string is not localized

File: app/src/main/java/com/example/kitchenflow/ui/main/OrdersAdapter.kt [hunk 6]
```diff
@@ -53,6 +55,43 @@ class OrdersAdapter(
...
+            } else {
+                "${((orderPrepTime - currentTime) / 1000 / 60).toInt()}m"
+            }
+        } else {
+            orderPrepTime -= MIN_DRIVE_TIME_MILLIS
+            ioTimerTv.text = if (currentTime >= orderPrepTime) {
+                ioTimerTv.setTextColor(context.getColor(R.color.colorPrimaryRed))
+                order.timer = ((currentTime - orderPrepTime) / 1000 / 60).toInt()
```
Comment (NikitaLychkovski) [inline app/src/main/java/com/example/kitchenflow/ui/main/OrdersAdapter.kt:97]: check `java.util.concurrent.TimeUnit` it is much more readable to use TimeUnit.MINUTES.toMillis(1) for example

File: app/src/main/res/layout/main_fragment.xml [hunk 7]
```diff
@@ -112,9 +113,10 @@
             android:layout_width="wrap_content"
             android:layout_height="wrap_content"
             android:layout_marginTop="32dp"
-            android:text="@string/current_orders"
+            android:text="@{`Current orders(` + model.currentOrders + `)`}"
```
Comment (NikitaLychkovski) [inline app/src/main/res/layout/main_fragment.xml]: Hardcoded string value

Comment (NikitaLychkovski) [reply app/src/main/res/layout/main_fragment.xml]: Also looks as null on UI
image

File: app/src/main/java/com/example/kitchenflow/ui/main/MainFragment.kt [hunk 8]
```diff
@@ -68,6 +82,24 @@ class MainFragment : Fragment() {
             }
             adapter = OrdersAdapter(requireContext())
             ordersListRv.adapter = adapter
+            searchBarEt.addTextChangedListener(object : TextWatcher {
```
Comment (NikitaLychkovski) [inline app/src/main/java/com/example/kitchenflow/ui/main/MainFragment.kt:85]: Before I suggested moving the entire TextWatcher to ViewModel. You can do it by custom databinding property

File: app/src/main/java/com/example/kitchenflow/ui/main/MainViewModel.kt [hunk 9]
```diff
@@ -1,16 +1,26 @@
...
 import com.example.kitchenflow.data.repository.KitchenRepository
 import io.reactivex.rxjava3.android.schedulers.AndroidSchedulers
 import io.reactivex.rxjava3.schedulers.Schedulers
+import java.text.SimpleDateFormat
+import java.util.*
 
 class MainViewModel(private val repository: KitchenRepository) : ViewModel() {
     private val ordersData: MutableLiveData<List<OrderModel>> = MutableLiveData()
     val orders: LiveData<List<OrderModel>> = ordersData
+    val date: ObservableField<String> = ObservableField(
```
Comment (NikitaLychkovski) [inline app/src/main/java/com/example/kitchenflow/ui/main/MainViewModel.kt:18]: Why observable if it set only once? If you are planing to update it in future put 
`SimpleDateFormat("EEE, MMMM dd", Locale.ENGLISH).format(
            Calendar.getInstance().time
        )` to function

File: app/src/main/res/layout/main_fragment.xml (hunk 7 above)
Comment (AntonGerasimovich) [reply app/src/main/res/layout/main_fragment.xml]: Are you sure your branch is updated? This problem is already fixed.

**Pull request merged**